src/helpers/helpers.core.js

```diff
@@ -283,22 +283,24 @@ export function _deprecated(scope, value, previous, current) {
   }
 }
 
+function indexOfDotOrLength(key, start) {
+  const idx = key.indexOf('.', start);
+  return idx === -1 ? key.length : idx;
+}
+
 export function resolveObjectKey(obj, key) {
-  // Special cases for `x` and `y` keys. It's quite a lot faster to aceess this way.
-  // Those are the default keys Chart.js is resolving, so it makes sense to be fast.
-  if (key === 'x') {
-    return obj.x;
-  }
-  if (key === 'y') {
-    return obj.y;
+  if (!key) {
+    return obj;
   }
-  const keys = key.split('.');
-  for (let i = 0, n = keys.length; i < n && obj; ++i) {
-    const k = keys[i];
-    if (!k) {
+  let pos = 0;
+  let idx = indexOfDotOrLength(key, pos);
+  while (idx > pos) {
+    obj = obj[key.substr(pos, idx - pos)];
+    if (!obj) {
```

Comment:
This previous check was on the key being defined/non-empty rather than the object. Perhaps the cause of the test failure?

Reply:
067791e got this case

Reply:
Yes, the falsy values returned the object in this new version. Comparing against empty string seems to make it a tiny bit slower:

![image](https://user-images.githubusercontent.com/27971921/107990979-4bcbc380-6fde-11eb-8dd8-5e78e663600c.png)

(could be just variation)